marlowe-playground-client/src/Marlowe/Linter.purs

```diff
@@ -239,10 +240,14 @@ constToVal x = Term (Constant x) { row: 0, column: 0 }
 -- | The aim here is to only traverse the contract once since we are concerned about performance with the linting
 -- FIXME: There is a bug where if you create holes with the same name in different When blocks they are missing from
 -- the final lint result. After debugging it's strange because they seem to exist in intermediate states.
-lint :: Term Contract -> State
-lint contract = state
+lint :: S.State -> Term Contract -> State
+lint contractState contract = state
```

Comment:
As a reader, I'd prefer it if this said `lint :: Semantics.State -> Term Contract -> State`. (ie., don't abbreviate the package name in this case.)